paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/WorldChunkMetaManager.java

```diff
@@ -41,8 +44,7 @@ public class WorldChunkMetaManager {
 	 */
 	private final Set<ChunkCoord> unloadingQueue;
 	private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
The two threadpool together may introduce a race condition. The unloading doesn't lock or signify the persisting of data, and the scheduled unloader may as a result persist and change data that is non-factual at this point. This needs a thorough review about the caching semantics and how the two thread pools interact with each other, or how other threads interact with this object.
Another check needs to be done if this object is the only source of truth for the ChunkCoord.

Reply:
If there is a race condition, then it existed before this change.

Reply:
But simply commiting another layer of asynchronicity over existing race conditions isn't going to fix them, just hide or make them worse in my opinion.